### PR TITLE
Channel Growth Feature (Backend)

### PR DESCRIPTION
**Pull request by eshleebien**
-grouped views and subscribers by channels and month
-added https server
-save results to mongo

### CODE REVIEW

**Comment by ninnzz** [comment]
Please use a descriptive message.
"Something new" is too vague.

File: controllers/channel.js [hunk 1]
```diff
@@ -1,44 +1,51 @@
...
+        var query = 'SELECT COUNT(channel_id) AS aggregate'
+                    + ' FROM mcn_channels'
+                    + ' WHERE (linked_date != "" OR linked_date IS NOT null)'
+                    + ' AND network != "" AND is_terminated != 1';
 
-        if (+req.query.page && +req.query.page > 0) {
-            page = +req.query.page;
-        }
+        data.page = (+data.page > 0) ? +data.page: 1;
+        data.limit = (+data.limit < 51 && +data.limit) ? +data.limit: 50;
```
Comment (wildtron) [inline controllers/channel.js:19]: `(+data.limit < 51) ? ...`

I think that would suffice just like what you did on [line 18](https://github.com/anyTV/channel-growth-feature/pull/3/files#diff-dc6ed861d77b3279c3e1c25809b15c93R18)

File: controllers/growth.js [hunk 2]
```diff
@@ -1,47 +1,171 @@
...
+            statistics = [],
+            statistic = [],
+            response = { results: [], channels: channels };
+
+        if (err) {
+            return next(err);
+        }
+
+        if (!result.length) {
+            return res.send({ results: [] });
```
Comment (wildtron) [inline controllers/growth.js:117]: why is this not sending back `response` instead?

File: controllers/channel.js (hunk 1 above)
Comment (eshleebien) [inline controllers/channel.js:19]: resolved

File: controllers/growth.js (hunk 2 above)
Comment (eshleebien) [inline controllers/growth.js:117]: resolved

**Pull request merged**